src/lib/converter/plugins/CommentPlugin.ts

```diff
@@ -314,11 +312,30 @@ export class CommentPlugin extends ConverterComponent {
         }
     }
 
+    /**
+     * Remove the specified reflections from the project.
+     */
+    static removeReflections(project: ProjectReflection, reflections: Reflection[]) {
+        const deletedIds = new Array<number>();
+        reflections.forEach((reflection) => {
+            CommentPlugin.removeReflection(project, reflection, deletedIds);
+        });
+
+        for (let key in project.symbolMapping) {
+            if (project.symbolMapping.hasOwnProperty(key) && deletedIds.indexOf(project.symbolMapping[key]) !== -1) {
+                delete project.symbolMapping[key];
+            }
+        }
+    }
+
     /**
      * Remove the given reflection from the project.
      */
-    static removeReflection(project: ProjectReflection, reflection: Reflection) {
-        reflection.traverse((child) => CommentPlugin.removeReflection(project, child));
+    private static removeReflection(project: ProjectReflection, reflection: Reflection, deletedIds: number[]) {
+        // keep track of the reflections that have been deleted
+        deletedIds.push(reflection.id);
```

Comment:
I'd move this to after the delete just to make sure the delete actually took place.